protocols/ping/src/handler.rs

```diff
@@ -271,6 +274,12 @@ impl ConnectionHandler for Handler {
         }
 
         loop {
+            if let Some(delay) = self.failure_delay.as_mut() {
+                if delay.poll_unpin(cx).is_pending() {
+                    return Poll::Pending;
+                }
+            }
```

Comment:
Instead of doing this here, I think we should just repeat line 321 - 330 for the `None` case.

Reply:
Just for clarity, are you suggesting that the `interval` be a `Option<Delay>`?

Reply:
I pushed a commit, let me know if you agree that this solves the problem :)